doc/CONFIGURATION.md

```diff
@@ -0,0 +1,289 @@
+# Configuration
+
+* [Overview](#overview)
+* [Modules](#modules)
+  * [Transport](#transport)
+  * [Stream Multiplexing](#stream-multiplexing)
+  * [Connection Encryption](#connection-encryption)
+  * [Peer Discovery](#peer-discovery)
+  * [Content Routing](#content-routing)
+  * [Peer Routing](#peer-routing)
+  * [DHT](#dht)
+  * [Pubsub](#pubsub)
+* [Libp2p node](#libp2p-node)
+* [Configuration examples](#configuration-examples)
+
+## Overview
+
+Considering libp2p's modular nature, it can be composed by a set of building blocks that enable users to create any type of p2p systems without needing to implement its network layer. Taking into account that each p2p system has its own needs and requirements, libp2p can be configured accordingly.
+
+`js-libp2p` acts as the composer for this modular p2p networking stack using libp2p compatible modules as its building blocks.
```

Comment:
I think this is superfluous. 
```suggestion
```

Reply:
Maybe I should move this under the next section? Its main intend is to introduce the `modules`